mermithid/processors/misc/MultiGasComplexLineShape.py

```diff
@@ -0,0 +1,3748 @@
+from morpho.processors import BaseProcessor
+from mermithid.misc import Constants, ComplexLineShapeUtilities, ConversionFunctions
+
+logger = morphologging.getLogger(__name__)
+
+
+
+__all__ = []
+__all__.append(__name__)
+
+class MultiGasComplexLineShape(BaseProcessor):
+
+    def InternalConfigure(self, params):
+        '''
+        Configure
+        '''
+        # Read other parameters
+        self.bins_choice = reader.read_param(params, 'bins_choice', [])
+        self.gases = reader.read_param(params, 'gases', ["H2", "Kr", "He", "Ar"])
+        # when self.fix_gas_composition and self.fix_width_scale_factor are both True,
+        # fit_data_simulated_resolution_scaled_fit_scatter_peak_ratio_with_fixed_gas_composition_and_width_scale_factor is used,
+        # Then the first N-1 gas compositions are set below through self.scatter_fractions_for_gases
+        # Otherwise, fit_data_simulated_resolution_scaled_fit_scatter_peak_ratio is used
+        self.fix_gas_composition = reader.read_param(params, 'fix_gas_composition', False)
+        self.fix_width_scale_factor = reader.read_param(params, 'fix_width_scale_factor', False)
+        self.scatter_fractions_for_gases = reader.read_param(params, 'scatter_fractions_for_gases', [])
+        self.max_scatters = reader.read_param(params, 'max_scatters', 20)
+        self.trap_weights = reader.read_param(params, 'trap_weights', {'weights':[0.076,  0.341, 0.381, 0.203], 'errors':[0.003, 0.013, 0.014, 0.02]}) #Weights from Xueying's Sept. 13 slides; errors currently arbitrary
+        self.fixed_scatter_proportion = reader.read_param(params, 'fixed_scatter_proportion', True)
+        if self.fixed_scatter_proportion == True:
+            self.scatter_proportion = reader.read_param(params, 'gas_scatter_proportion', [])
+        self.partially_fixed_scatter_proportion = reader.read_param(params, 'partially_fixed_scatter_proportion', True)
+        if self.partially_fixed_scatter_proportion == True:
+            self.free_gases = reader.read_param(params, 'free_gases', ["H2", "He"])
+            self.fixed_gases = reader.read_param(params, 'fixed_gases', ["Ar", "Kr"])
+            self.gases = self.free_gases + self.fixed_gases
+            self.scatter_proportion_for_fixed_gases = reader.read_param(params, 'scatter_proportion_for_fixed_gases', [0.018, 0.039])
+        self.fixed_survival_probability = reader.read_param(params, 'fixed_survival_probability', True)
+        if self.fixed_survival_probability == True:
+            self.survival_prob = reader.read_param(params, 'survival_prob', 1)
+        self.use_radiation_loss = reader.read_param(params, 'use_radiation_loss', True)
+        self.sample_ins_resolution_errors = reader.read_param(params, 'sample_ins_res_errors', False)
+        # configure the resolution functions: gaussian_lorentzian_composite_resolution, elevated_gaussian, composite_gaussian, composite_gaussian_pedestal_factor, and simulated_resolution_scaled
```

Comment:
add a list of resolutions to the processor description also.

Reply:
Added the list of resolutions to the processor description.